sysadmin/configs/nginx/default-integration

```diff
@@ -19,7 +19,10 @@ server {
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 
         location /demo/clientApp/ {
-            expires 1m;
+            # Request the browser not to cache
+            add_header Cache-Control "no-store, max-age=-1";
```

Comment:
By using no-store , we will not cache/store any of the response in the browser. That means, the user will always download the files .

What we want, if there is an update the navigator will check if there is any difference between current cache (version file) and the ones from server .
For that we need to use no-cache , which will ask every request to the serve if it can show the user's cache (with ETag).
If we add an expiration, we can use must-revalidate , once it expired it will ask the server and revalidate .

These schema will illustrate the different way to do . 

![image](https://github.com/ipol-journal/ipolDevel/assets/101263939/d8f4372b-2cc3-43ba-aa70-fcedf6531973)

Reply:
> By using no-store , we will not cache/store any of the response in the browser. That means, the user will always download the files .

I've configured it this way because we don't want to have any cache issues when testing in Integration. It's the same configuration for local. However, in prod it requests the client to cache.

Reply:
Well, let's see if we reach a consensus.
I propose this:
- Local: no cache at all.
- Integration: very small cache, like 5 seconds
- Production: normal caching.

Reply:
I think the cache should be identical between integration and production, otherwise what's the point of integration?

Reply:
We need to find a balance. Indeed, having integration=production allows to very well approximate the environments. However, Integration is also used for our testing and it's useful that changes are available immediately.
But if think it's better to put the same invalidation time as in production, I'm fine with that.

Reply:
In the diagram above, there is a branch "Must check before using cached copy? -> Always"
If we used this path, there would be no issue of invalidation time, right? Both for production and integration.